spring-cloud-commons/src/main/java/org/springframework/cloud/client/hedger/HedgingClient.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.client.hedger;
+
+import java.time.Duration;
+
+import org.springframework.lang.Nullable;
+import org.springframework.web.reactive.function.client.ClientRequest;
+import org.springframework.web.reactive.function.client.ClientResponse;
+
+public interface HedgingClient {
+	boolean shouldHedge(ClientRequest request);
+
+	int getNumberOfHedgedRequests(ClientRequest request);
+
+	Duration getDelayBeforeHedging(ClientRequest request);
+
+	void record(ClientRequest request,
```

Comment:
Do we need this here? It seems that `HedgingMetricsReporter` already has this

Reply:
This is the feedback loop to get the latency of the request to update the 95th percentile.

When you originally implemented hedging, you did it via two separate `ExchangeFilterFunction` objects: one to record the latency, and one to calculate the 95th percentile and do the hedging. Putting this here is what enables this to become just a single function.

Reply:
It does feel like it makes sense to separate getting the hedge parameters and recording. Some hedge policies might not even care about recording the latencies, maybe they just want to use fixed delay.